### PR TITLE
JS Components: Add QRCode component

### PR DESCRIPTION
**Pull request by retrofox**
Fixes https://github.com/Automattic/jetpack/issues/23272

#### Changes proposed in this Pull Request:

* JS Components: Add ~QRPost~ QRCode component

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Test with Storybook
```cli
cd ./projects/js-packages/storybook
```

```cli
pnpm install & pnpm storybook:dev
```

A browser tab should be opened with the `http://localhost:50240` address.

* Activate the QRPost story: `http://localhost:50240/?path=/story/js-packages-components-block-editor-qr-code--default`
* Play with the Addons tool to change component properties ~less the `value` one. It isn't possible to change it because it's internally handled by the store hooks.~

### CODE REVIEW

File: projects/js-packages/components/components/qr-post/index.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
```
Comment (jeherve) [inline projects/js-packages/components/components/qr-post/index.jsx:4]: Could we get rid of the react imports if we use this in the editor context with WordPress elements?

File: projects/js-packages/components/components/qr-post/index.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	const {
+		post: { title: postTitle },
+		permalink,
+		edits: { title: editTitle },
+	} = useSelect(
+		select => ( {
+			post: select( editorStore ).getCurrentPost(),
+			slug: select( editorStore ).getEditedPostSlug(),
+			permalink: select( editorStore ).getPermalink(),
+			edits: select( editorStore ).getPostEdits(),
```
Comment (jeherve) [inline projects/js-packages/components/components/qr-post/index.jsx:42]: What's the advantage of using `store` here versus doing something like this?

```suggestion
			post: select( 'core/editor' ).getCurrentPost(),
			slug: select( 'core/editor' ).getEditedPostSlug(),
			permalink: select( 'core/editor' ).getPermalink(),
			edits: select( 'core/editor' ).getPostEdits(),
```

**Review by jeherve — CHANGES_REQUESTED** [review]
As per our Slack discussion, it may make more sense to keep this closer to the actual plugin in `projects/plugins/jetpack/extensions/plugins` for now, until we need it for other projects?
p1646675365223949-slack-CBG1CP4EN

File: projects/js-packages/components/components/qr-post/index.jsx (hunk 2 above)
Comment (retrofox) [reply projects/js-packages/components/components/qr-post/index.jsx]: If I'm not wrong, it's part of what the core API suggests to deal with it. I supposed that, if for some reason the store id needs to change there,   it won't impact the code here.

**Review by oskosk — APPROVED** [review]
LGTM!

**Pull request merged**